### PR TITLE
feat(activity-logging): UI lets you save with no changes. Ignore those updates

### PR DESCRIPTION
**Pull request by pauldambra**
…

## Problem

You can save in the UI without changing anything. This causes us to log an activity that the UI will never display.

## Changes

this ignores activities labelled "updated" that have no changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

added a unit test
and saved in the UI without changes to verify an activity isn't logged any more by checking the DB

### CODE REVIEW

File: posthog/models/activity_logging/activity_log.py [hunk 1]
```diff
@@ -101,6 +104,16 @@ def log_activity(
...
     detail: Detail,
 ) -> None:
+    if activity == "updated" and (detail.changes is None or len(detail.changes) == 0):
+        logger.warn(
+            "'updated' activity was logged with no changes and was ignored",
+            team_id=team_id,
+            organization_id=organization_id,
+            user_id=user.id,
+            scope=scope,
+        )
```
Comment (Twixes) [inline posthog/models/activity_logging/activity_log.py:114]: Logging here seems like it'd generate quite a bit of logs that wouldn't really ever be used, so I'm wondering if we should maybe drop it… CC @guidoiaquinti

**Review by Twixes — COMMENTED** [review]
+1 to the change, only not sure if there's a benefit to logging here

File: posthog/models/activity_logging/activity_log.py (hunk 1 above)
Comment (guidoiaquinti) [reply posthog/models/activity_logging/activity_log.py:114]: What's the expected event of volumes? If it's a lot and we don't need high cardinality data we could also use a metric that is cheaper.

In any case: for better parsing `"'updated' activity was logged with no changes and was ignored",` -> `ignore_update_activity_change`

Comment (mariusandra) [reply posthog/models/activity_logging/activity_log.py:114]: When standardising the interface, I think we should move towards "save" buttons that are disabled when there's nothing to save, like in the new insight interface. Hence the traffic should drop to near zero eventually.

Comment (pauldambra) [reply posthog/models/activity_logging/activity_log.py:114]: very unscientifically it is about 1 in 1500

We had about 1400 feature flag changes over a weekend, one of those had an empty array.

It's a weak debug signal anyway, so I don't mind removing it...

Comment (guidoiaquinti) [reply posthog/models/activity_logging/activity_log.py:114]: Let merge this and keep an 👁️ on the volume

File: posthog/models/activity_logging/activity_log.py [hunk 2]
```diff
@@ -101,6 +104,16 @@ def log_activity(
     activity: str,
     detail: Detail,
 ) -> None:
+    if activity == "updated" and (detail.changes is None or len(detail.changes) == 0):
+        logger.warn(
+            "'updated' activity was logged with no changes and was ignored",
```
Comment (pauldambra) [inline posthog/models/activity_logging/activity_log.py:109]:
```suggestion
            "ignore_update_activity_no_changes",
```

**Review by Twixes — APPROVED** [review]

**Pull request merged**